homeassistant/components/sense/__init__.py

```diff
@@ -57,7 +55,17 @@ def get_device_by_id(self, sense_device_id):
         return self._data_by_device.get(sense_device_id)
 
 
-async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+@dataclass
```

Comment:
```suggestion
@dataclass(slots=True)
```

Reply:
Ah, I've been meaning to ask you if this improves performance. Looks like it should.